pkg/controller/instance/instance_controller.go

```diff
@@ -86,27 +87,17 @@ func (r *Reconciler) SetupWithManager(
 			return requests
 		})
 
-	// hasAnnotation returns true if an annotation with the passed key is found in the map
-	hasAnnotation := func(key string, annotations map[string]string) bool {
-		if annotations == nil {
-			return false
-		}
-		for k := range annotations {
-			if k == key {
-				return true
-			}
-		}
-		return false
-	}
-
 	// resPredicate ignores DeleteEvents for pipe-pods only (marked with task.PipePodAnnotation). This is due to an
 	// inherent race that was described in detail in #1116 (https://github.com/kudobuilder/kudo/issues/1116)
-	// tl;dr: pipe task will delete the pipe pod at the end of the execution. this would normally trigger another
+	// tl;dr: pipe-task will delete the pipe pod at the end of the execution. this would normally trigger another
 	// Instance reconciliation which might end up copying pipe files twice. we avoid this by explicitly ignoring
 	// DeleteEvents for pipe-pods.
 	resPredicate := predicate.Funcs{
-		CreateFunc:  func(event.CreateEvent) bool { return true },
-		DeleteFunc:  func(e event.DeleteEvent) bool { return !hasAnnotation(task.PipePodAnnotation, e.Meta.GetAnnotations()) },
+		CreateFunc: func(event.CreateEvent) bool { return true },
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return e.Meta.GetAnnotations() != nil &&
+				funk.Contains(e.Meta.GetAnnotations(), task.PipePodAnnotation)
```

Comment:
Using the untyped method `Contains` since annotations is a `map[string]string` and has no typed implementation.

Reply:
@zen-dog catching up, so is this using `reflect` then?

Reply:
Welcome back, @gerred ;) yes, the generic methods e.g. `Contains` use reflection to detect the correct type. It is yet to be proved, but I suspect that the majority of our use cases could be covered by the typed methods e.g. `ContainsString`.